Can't make <ol> to render

Hi @yhatt - I'm facing an issue with the `<ol>` element, I wonder if you could help me? It looks to me like a bug related to `jsx-slack`. 

Here is what I'm trying to do:
```
await client.chat.postMessage({
      token: botToken,
      channel,
      text: 'Sorry about that. Please report an issue to your administrator.',
      blocks: JSXSlack(
        <Blocks>
          <Section>
            <ol type="1">
              <li>First item</li>
              <li>Second item</li>
            </ol>
          </Section>
        </Blocks>
      )
    });
```
and I'm getting the following error:
```
[ERROR]  bolt-app ERROR: TypeError: Cannot convert object to primitive value
```

Going further, if I remove the second item, and leave just the first one, I'm getting following error:
```
[ERROR]  bolt-app ERROR: TypeError: u[t].toString is not a function
```

If I rename `<ol>` to `<ul>` (and remove `type` of course), I'm getting properly displayed unordered list in Slack. So the problem seems to be only with an ordered list. The other elements and blocks are rendered fine. 
Note: We recently switched from `@speee-js/jsx-slack` to `jsx-slack` - not sure if this is when an issue with `ol` started to happen, but I think it could be the case.

The project is written in typescript, here is `tsconfig.json`: 
```
{
  "compilerOptions": {
    "target": "ESNext",
    "module": "commonjs",
    "lib": ["dom", "es6"],
    "allowJs": true,
    "jsx": "react-jsx",
    "jsxImportSource": "jsx-slack",
    "isolatedModules": false,
    "strict": true,
    "noImplicitAny": true,
    "moduleResolution": "node",
    "baseUrl": "src",
    "outDir": "dist",
    "esModuleInterop": true,
    "resolveJsonModule": true,
    "sourceMap": true,
    "paths": {
      "@/*": ["./*"]
    }
  },
  "include": ["src"],
  "exclude": ["dist", "node_modules"]
}
```

and `package.json`:
```
"dependencies": {
    "@sendgrid/mail": "^7.5.0",
    "@slack/bolt": "^3.8.1",
    "@types/mocha": "^8.2.0",
    "@types/node": "^14.17.27",
    "@types/puppeteer": "^5.4.4",
    "axios": "^0.24.0",
    "date-fns": "^2.25.0",
    "dotenv": "^8.2.0",
    "jsx-slack": "^4.5.2",
    "mixpanel": "^0.14.0",
    "module-alias": "^2.2.2",
    "puppeteer": "^10.4.0",
    "typescript": "4.5.2"
  },
  "devDependencies": {
    "@commitlint/cli": "^8.3.5",
    "@commitlint/config-conventional": "^8.3.4",
    "@typescript-eslint/eslint-plugin": "^3.2.0",
    "@typescript-eslint/parser": "^3.2.0",
    "assert": "^2.0.0",
    "eslint": "^7.2.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-prettier": "^3.1.3",
    "husky": "^4.2.5",
    "lint-staged": "^10.2.9",
    "mocha": "^8.3.0",
    "prettier": "^2.0.5",
    "ts-node": "^8.10.2",
    "ts-node-dev": "^1.0.0-pre.44",
    "ts-node-paths": "^1.0.1"
  },
  "engines": {
    "node": "14.x"
  }
```

Thanks!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+### Fixed
+
+- Fix `TypeError` while rendering lists and hyperlinks with JSX runtime ([#258](https://github.com/yhatt/jsx-slack/issues/258), [#259](https://github.com/yhatt/jsx-slack/pull/259))
+
 ## v4.5.2 - 2021-12-18
 
 ### Added

--- a/jest.config.js
+++ b/jest.config.js
@@ -46,5 +46,6 @@ module.exports = {
   restoreMocks: true,
   testEnvironment: 'node',
   testMatch: ['<rootDir>/test/**/!(_)*.[jt]s?(x)'],
+  testPathIgnorePatterns: ['/node_modules/', 'babel.config.js'],
   transformIgnorePatterns: [`/node_modules/(?!${esModules.join('|')})`],
 }

--- a/src/mrkdwn/jsx.ts
+++ b/src/mrkdwn/jsx.ts
@@ -13,7 +13,12 @@ const buildAttr = (props: { [key: string]: any }, escapeEntities = true) => {
   let attr = ''
 
   for (const prop of Object.keys(props)) {
-    if (props[prop] !== undefined) {
+    if (
+      props[prop] != null &&
+      ['number', 'bigint', 'boolean', 'string', 'symbol'].includes(
+        typeof props[prop]
+      )
+    ) {
       let attrBase = props[prop].toString()
       if (escapeEntities) attrBase = escapeEntity(attrBase)
 

--- a/test/babel/automatic.jsx
+++ b/test/babel/automatic.jsx
@@ -1,6 +1,6 @@
 /** @jsxImportSource jsx-slack */
 import { JSXSlackError } from '../../src/error'
-import { Blocks, Fragment, Section, Select } from '../../src/index'
+import { Blocks, Fragment, Mrkdwn, Section, Select } from '../../src/index'
 
 jest.mock('../../jsx-dev-runtime')
 
@@ -13,17 +13,48 @@ describe('Babel transpilation through automatic runtime (Development mode)', () 
         </Section>
       </Blocks>
     ).toMatchInlineSnapshot(`
-    Array [
+          Array [
+            Object {
+              "text": Object {
+                "text": "Hello, world!",
+                "type": "mrkdwn",
+                "verbatim": true,
+              },
+              "type": "section",
+            },
+          ]
+      `)
+  })
+
+  it('accepts list items', () => {
+    expect(
+      <Mrkdwn>
+        <ul>
+          <li>item</li>
+        </ul>
+      </Mrkdwn>
+    ).toMatchInlineSnapshot(`
       Object {
-        "text": Object {
-          "text": "Hello, world!",
-          "type": "mrkdwn",
-          "verbatim": true,
-        },
-        "type": "section",
-      },
-    ]
-  `)
+        "text": "• item",
+        "type": "mrkdwn",
+      }
+    `)
+
+    // An edge case has reported to ordered list in https://github.com/yhatt/jsx-slack/issues/258)
+    expect(
+      <Mrkdwn>
+        <ol>
+          <li>item</li>
+          <li>item</li>
+        </ol>
+      </Mrkdwn>
+    ).toMatchInlineSnapshot(`
+      Object {
+        "text": "1. item
+      2. item",
+        "type": "mrkdwn",
+      }
+    `)
   })
 
   it('accepts fragment syntax', () => {
@@ -42,33 +73,33 @@ describe('Babel transpilation through automatic runtime (Development mode)', () 
         <Component />
       </Blocks>
     ).toMatchInlineSnapshot(`
-    Array [
-      Object {
-        "text": Object {
-          "text": "Section A",
-          "type": "mrkdwn",
-          "verbatim": true,
-        },
-        "type": "section",
-      },
-      Object {
-        "text": Object {
-          "text": "Section B",
-          "type": "mrkdwn",
-          "verbatim": true,
-        },
-        "type": "section",
-      },
-      Object {
-        "text": Object {
-          "text": "Section C",
-          "type": "mrkdwn",
-          "verbatim": true,
-        },
-        "type": "section",
-      },
-    ]
-  `)
+          Array [
+            Object {
+              "text": Object {
+                "text": "Section A",
+                "type": "mrkdwn",
+                "verbatim": true,
+              },
+              "type": "section",
+            },
+            Object {
+              "text": Object {
+                "text": "Section B",
+                "type": "mrkdwn",
+                "verbatim": true,
+              },
+              "type": "section",
+            },
+            Object {
+              "text": Object {
+                "text": "Section C",
+                "type": "mrkdwn",
+                "verbatim": true,
+              },
+              "type": "section",
+            },
+          ]
+      `)
 
     expect(fragment.$$jsxslack.type).toBe(Fragment)
   })

--- a/test/babel/babel.config.js
+++ b/test/babel/babel.config.js
@@ -1,0 +1,3 @@
+// This is workaround for https://github.com/facebook/jest/issues/12000.
+// Please remove if Jest was fixed to parse babel config placed to the root.
+module.exports = require('../../babel.config')

